src/group_cmd.cpp

```diff
@@ -664,42 +665,48 @@ CommandCost CmdSetGroupLivery(TileIndex tile, DoCommandFlag flags, uint32 p1, ui
 }
 
 /**
- * Set replace protection for a group and its sub-groups.
+ * Set group flag for a group and its sub-groups.
  * @param g initial group.
- * @param protect 1 to set or 0 to clear protection.
+ * @param set 1 to set or 0 to clear protection.
  */
-static void SetGroupReplaceProtection(Group *g, bool protect)
+static void SetGroupFlag(Group *g, GroupFlags flag, bool set, bool children)
 {
-	g->replace_protection = protect;
+	if (set) {
+		SetBit(g->flags, flag);
+	} else {
+		ClrBit(g->flags, flag);
+	}
+
+	if (!children) return;
 
 	for (Group *pg : Group::Iterate()) {
-		if (pg->parent == g->index) SetGroupReplaceProtection(pg, protect);
+		if (pg->parent == g->index) SetGroupFlag(pg, flag, set, true);
 	}
 }
 
 /**
- * (Un)set global replace protection from a group
+ * (Un)set group flag from a group
  * @param tile unused
  * @param flags type of operation
  * @param p1   index of group array
- * - p1 bit 0-15 : GroupID
+ * - p1 bit 0-15  : GroupID
+ * - p1 bit 16-18 : Flag to set
  * @param p2
  * - p2 bit 0    : 1 to set or 0 to clear protection.
  * - p2 bit 1    : 1 to apply to sub-groups.
  * @param text unused
  * @return the cost of this operation or an error
  */
-CommandCost CmdSetGroupReplaceProtection(TileIndex tile, DoCommandFlag flags, uint32 p1, uint32 p2, const char *text)
+CommandCost CmdSetGroupFlag(TileIndex tile, DoCommandFlag flags, uint32 p1, uint32 p2, const char *text)
 {
-	Group *g = Group::GetIfValid(p1);
+	Group *g = Group::GetIfValid(GB(p1, 0, 16));
 	if (g == nullptr || g->owner != _current_company) return CMD_ERROR;
 
+	GroupFlags flag = (GroupFlags)GB(p1, 16, 3);
```

Comment:
It doesn't look like the command is short on bits, so just extract the full 8 bit to avoid potential future oversight?

Reply:
The storage is 8 bits as a bit field, but flag is sent over as a value rather than bit. So 3 bits corresponds to the 8 possible values of the 8 bit storage.